dev/ProgressRing/ProgressRingLoading.cpp

```diff
@@ -0,0 +1,362 @@
+    // Ellipse Path
+    winrt::CompositionSpriteShape SpriteShape_3()
+    {
+        auto result = _c.CreateSpriteShape();
+        result.Geometry(Ellipse_7_2());
+        result.StrokeBrush(AnimatedColorBrush_AlmostDodgerBlue_FF0078D7_to_TransparentAlmostDodgerBlue_000078D7());
+        result.StrokeDashCap(winrt::CompositionStrokeCap::Round);
+        result.StrokeEndCap(winrt::CompositionStrokeCap::Round);
+        result.StrokeStartCap(winrt::CompositionStrokeCap::Round);
+        result.IsStrokeNonScaling(true);
+        result.StrokeThickness(static_cast<float>(_strokeThickness));
+        return result;
+    }
+
+    // Layer (Shape): Radial
+    // Transforms: Radial
+    // Ellipse Path
+    // Ellipse Path.EllipseGeometry
+    // TrimStart
+    winrt::StepEasingFunction StepThenHoldEasingFunction()
+    {
+        auto result = _c.CreateStepEasingFunction();
+        result.IsInitialStepSingleFrame(true);
+        return result;
+    }
+
+    // Layer (Shape): Radial
+    // Transforms: Radial
+    // Ellipse Path
+    // Ellipse Path.EllipseGeometry
+    // TrimEnd
+    winrt::ScalarKeyFrameAnimation TrimEndScalarAnimation_0_to_0p5()
+    {
+        auto result = _c.CreateScalarKeyFrameAnimation();
+        result.Duration({ c_durationTicks });
+        result.InsertKeyFrame(0, 9.99999975E-05F, _holdThenStepEasingFunction);
+        result.InsertKeyFrame(0.5F, 0.5F, _cubicBezierEasingFunction);
+        return result;
+    }
+
+    // Layer (Shape): Radial
+    // Transforms: Radial
+    // Ellipse Path
+    // Ellipse Path.EllipseGeometry
+    // TrimStart
+    winrt::ScalarKeyFrameAnimation TrimStartScalarAnimation_0_to_0p5()
+    {
+        auto result = _c.CreateScalarKeyFrameAnimation();
+        result.Duration({ c_durationTicks });
+        result.InsertKeyFrame(0, 0, StepThenHoldEasingFunction());
+        result.InsertKeyFrame(0.5F, 0, HoldThenStepEasingFunction());
+        result.InsertKeyFrame(1, 0.5F, CubicBezierEasingFunction());
+        return result;
+    }
+
+public:
+    ProgressRingAnimatedVisual(winrt::Compositor compositor, double strokeThickness, winrt::Size size, winrt::Color foreground, winrt::Color background)
+        : _c(compositor),
+        _reusableExpressionAnimation(compositor.CreateExpressionAnimation()),
+        _strokeThickness(strokeThickness),
+        _size(size),
+        _foreground(foreground),
+        _background(background)
+    {
+        const auto _ = Root();
+    }
+
+    [[nodiscard]] auto RootVisual() const
+    {
+        return _root;
+    }
+
+    [[nodiscard]] winrt::float2 Size() const
+    {
+        return _size;
+    }
+
+    [[nodiscard]] std::chrono::milliseconds Duration() const
+    {
+        return { c_durationTicks };
+    }
+
+    void Close()
+    {
+        if (_root)
+        {
+            _root.Close();
+        }
+    }
+};
+
+ProgressRingLoading::ProgressRingLoading(double strokeThickness, winrt::Size size, winrt::Color foreground, winrt::Color background)
```

Comment:
>double [](start = 41, length = 6)

Why not pass this in as a float so you can avoid the casts later on. The Composition APIs only accept floats.